collab-folder/src/folder.rs

```diff
@@ -596,7 +598,16 @@ impl FolderBody {
   /// # Returns
   ///
   /// * `Vec<View>`: A vector of `View` objects that includes the parent view and all of its child views.
-  pub fn get_view_recursively_with_txn<T: ReadTxn>(&self, txn: &T, view_id: &str) -> Vec<View> {
+  pub fn get_view_recursively_with_txn<T: ReadTxn>(
+    &self,
+    txn: &T,
+    view_id: &str,
+    mut visited: HashSet<String>,
+  ) -> Vec<View> {
```

Comment:
Technically since this method is recursive, it would be better to pass resulting `&mut Vec<View>` as accumulating argument instead of copying data over on each recursive step.

Reply:
Updated the method signature. Since there are significant number of changes now, i think it's better if i add some tests or test this end to end before merging.

Reply:
Technically you can still keep the public API unchanged, and move recursion params to an inner (private) method. This should make it - more or less - a drop-in replacement.